docs: Tyk Operator 1.4.0 release notes (#1826)
TykTechnologies/tyk-docs: developer-support/release-notes/operator.mdx

  - ```bash
    docker pull tykio/tyk-operator:v1.4.0
    ```
- Helm chart
  - [tyk-charts v5.2.0](/developer-support/release-notes/helm-chart#5-2-0-release-notes)

#### Changelog
<a id="Changelog-v1.4.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

<Accordion title='Add MCP Proxy support to TykMcpProxyDefinition and SecurityPolicy CRDs'>
This release introduces two extensions to Tyk Operator's custom resource support for MCP (Model Context Protocol) Gateway:

**TykMcpProxyDefinition CRD**

A new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxy API definitions declaratively in Kubernetes. The OAS definition is supplied via a ConfigMap reference (`spec.tykMCP.configmapRef`). The CRD status surface exposes three hashes — `CRDSpecHash`, `ConfigMapHash`, and `TykSpecHash` — to track synchronisation state between the Kubernetes resource, the ConfigMap, and the Tyk Gateway definition.

**SecurityPolicy CRD — MCP access rights**

The `SecurityPolicy` CRD has been extended to support MCP-specific access rights:
- Per-tool, per-resource, and per-prompt allow/deny lists (`mcp_access_rights`)
- Per-JSON-RPC-method allow/deny lists (`json_rpc_methods_access_rights`)

> - Per-primitive rate limits (`mcp_primitives`) and per-method rate limits (`json_rpc_methods`)

Reviewer: We need to update the k8s reference documentation for this @v008370-git

---

## 1.4 Release Notes

### 1.4.0 Release Notes

> #### Release Date 8 May 2026

Reviewer: ```suggestion
#### Release Date 19 May 2026
```

#### Release Highlights

> Tyk Operator 1.4.0 introduces support for managing MCP (Model Context Protocol) Proxy API definitions through Kubernetes custom resources. The new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxy APIs declaratively, and the `SecurityPolicy` CRD has been extended with MCP access rights for tool-based access control and per-primitive rate limiting.

Reviewer 2: ```suggestion
Tyk Operator 1.4.0 introduces support for managing [MCP (Model Context Protocol) Proxy](/ai-management/mcp-gateway/managing-mcp-proxies) definitions through Kubernetes custom resources. The new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxies declaratively, and the `SecurityPolicy` CRD has been extended with MCP access rights for tool-based access control and per-primitive rate limiting.
```

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.4.0).

<Note>

> The initial Helm chart for Operator 1.4.0 contained an installation bug. Please ensure you use tyk-charts version [5.2.0](/developer-support/release-notes/helm-chart#5-2-0-release-notes) or later to install Operator 1.4.0.

Reviewer 2: ```suggestion
The initial Helm chart for Operator 1.4.0 contained an installation bug. Please ensure you use [Tyk Charts 5.2.0](/developer-support/release-notes/helm-chart#5-2-0-release-notes) or later to install Operator 1.4.0.
```

Please do not use github repo names when referring to Tyk products.

What do we mean by "initial Helm chart"? What version?

Reviewer: It seems it is copied from previous release notes. @JoanCamosTyk Check if this is required.

</Note>

#### Breaking Changes
This release has no breaking changes.

#### Dependencies
##### 3rd Party Dependencies & Tools

| Third Party Dependency                                     | Tested Versions        | Compatible Versions    | Comments | 
| :---------------------------------------------------------- | :---------------------- | :---------------------- | :-------- | 
| [Kubernetes](https://kubernetes.io)                        | 1.28.x to 1.32.x       | 1.19.x to 1.32.x       |          | 

Given the time difference between your upgrade and the release of this version, we recommend customers verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

#### Deprecations
There are no deprecations in this release.

#### Upgrade instructions

> Deploy Tyk Operator 1.4.0 only after Tyk Gateway 5.13.0 and Tyk Dashboard 5.13.0 are running. The `TykMcpProxyDefinition` controller calls Dashboard's `/api/mcps` endpoint, which does not exist in earlier Dashboard versions. Deploying Operator before Dashboard and Gateway will cause reconciliation failures on all `TykMcpProxyDefinition` resources. See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for details.

Reviewer 2: ```suggestion
The MCP Proxy support in Tyk Operator 1.4.0 requires Tyk Gateway 5.13.0 and Tyk Dashboard 5.13.0 (or newer)

Attempting to use `TykMcpProxyDefinition` resources with Operator  1.4.0 prior to upgrading Gateway and Dashboard to a version supporting MCP Proxies will cause reconciliation failures.

See [MCP Gateway upgrade considerations](/developer-support/upgrading#mcp-upgrade-considerations) for details.
```

Go to the [Upgrading Tyk Operator](/tyk-stack/tyk-operator/installing-tyk-operator) section for detailed upgrade instructions.

#### Downloads
- [Docker image v1.4.0](https://hub.docker.com/r/tykio/tyk-operator/tags?page=&page_size=&ordering=&name=v1.4.0)
  - ```bash
    docker pull tykio/tyk-operator:v1.4.0
    ```
- Helm chart
  - [tyk-charts v5.2.0](/developer-support/release-notes/helm-chart#5-2-0-release-notes)

#### Changelog
<a id="Changelog-v1.4.0" data-scroll-offset></a>

##### Added

<AccordionGroup>

> <Accordion title='Add MCP Proxy support to TykMcpProxyDefinition and SecurityPolicy CRDs'>

Reviewer 2: ```suggestion
<Accordion title='Add MCP Proxy support'>
```

> This release introduces two extensions to Tyk Operator's custom resource support for MCP (Model Context Protocol) Gateway:

Reviewer 2: ```suggestion
This release adds support for MCP (Model Context Protocol) Proxies with a new TykMcpProxyDefinition CRD and an extension to the SecurityPolicy CRD.
```

**TykMcpProxyDefinition CRD**

> A new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxy API definitions declaratively in Kubernetes. The OAS definition is supplied via a ConfigMap reference (`spec.tykMCP.configmapRef`). The CRD status surface exposes three hashes — `CRDSpecHash`, `ConfigMapHash`, and `TykSpecHash` — to track synchronisation state between the Kubernetes resource, the ConfigMap, and the Tyk Gateway definition.

Reviewer 2: Why "OAS definition" (this actually has no meaning - presumably this isn't an OpenAPI document...) do you mean Tyk OAS API definition, or MCP API definition?

> A new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxy API definitions declaratively in Kubernetes. The OAS definition is supplied via a ConfigMap reference (`spec.tykMCP.configmapRef`). The CRD status surface exposes three hashes — `CRDSpecHash`, `ConfigMapHash`, and `TykSpecHash` — to track synchronisation state between the Kubernetes resource, the ConfigMap, and the Tyk Gateway definition.

Reviewer 2: ```suggestion
A new `TykMcpProxyDefinition` CRD allows you to manage MCP Proxy definitions declaratively in Kubernetes. 

The CRD status surface exposes three hashes — `CRDSpecHash`, `ConfigMapHash`, and `TykSpecHash` — to track synchronisation state between the Kubernetes resource, the ConfigMap, and the Tyk Gateway definition.
```
